### ISSUE TITLE
The documentation is not really clear about header assertions

### ISSUE THREAD

**Issue by legzo** [Orange-OpenSource/hurl#96]
Hi,

I have a test with :

```
GET http://someUrl
Accept-Encoding: gzip

HTTP/1.1 200
Content-Encoding: gzip
```

My server actually returns : 

```
Content-Encoding: gzip
Content-Encoding: gzip
```

Which is a bug, but that's another story... 

The assertion passes, and maybe it's OK.

But when I change the assertion to :

```
GET http://someUrl
Accept-Encoding: gzip

HTTP/1.1 200
Content-Encoding: "gzip"
```

It fails with 

```    
    |
 21 | Content-Encoding: "gzip"
    |                   ^^^^^^ actual value is <["gzip", "gzip"]>
```

Do you think that it is logical ? There is no mention of `Header: "expectedValue"`, only `Header: expectedValue`. I mean is the `Header: expectedValue` syntax just a `contains()` ?

Maybe just a clarification could be added...

Thanks

**Comment by fabricereix** [comment]
Enclosing the header value between quotes adds quotes to the value itself.

The typical example is the [ETag header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag)
`ETag: "<etag_value>"`

We can indeed clarify that in the [doc](https://hurl.dev/docs/asserting-response.html#headers)


The error message is a bit clearer if we receive only one Content-Encoding header.
```
   |
21 | Content-Encoding: "gzip"
   |                   ^^^^^^ actual value is <gzip>
```


Back to your example,
you receive two headers with value `<gzip>`, which is treated as a list of 2 strings <gzip>.
The error message could be maybe improved to:
```
    |
 21 | Content-Encoding: "gzip"
    |                   ^^^^^^ actual value is [<gzip>, <gzip>]
```

**Comment by legzo** [comment]
The fact is that I can't assert that I have the header without quotes but only once can I ?

Because now that I fixed my bug and my service returns the header only once, the check if quote fails (just as in your first snippet), but if my assertion is without quotes, it passes, but it passed also with two headers...

How would you check for only one header with a value without quotes ?

**Comment by fabricereix** [comment]
Indeed, an implicit header assert (defined directly after the status code) does not fail with duplicate headers.

In order to test that, you need to use the explicit header assert
```
[Asserts]
header "Content-Encoding" equals "gzip"
```

Current, removing an implicit header just remove a check.
If the hurl file passed before, it will pass after.
that's the reason why we choose to ignore the duplicate header issue.
This also works nicely with the Set-Cookie header.

**Comment by legzo** [comment]
Thanks !

**Comment by fabricereix** [comment]
doc on [headers](https://hurl.dev/docs/asserting-response.html#headers) has been updated
